src/pydata_sphinx_theme/__init__.py

```diff
@@ -169,6 +194,15 @@ def prepare_html_config(app, pagename, templatename, context, doctree):
     # update version number
     context["theme_version"] = __version__
 
+    # check if a permanent search filed is set. If yes, the search button will be
+    # hidden in the final layout
+    context["theme_with_search"] = False
```

Comment:
to me "theme_with_search" is not clear. Maybe something like "use_search_overlay"?  Or reverse the boolean and call it "use_search_field" or "persistent_search_field"?

Reply:
agreed